iOS Best Practises Additions (#4061)
circleci/circleci-docs: jekyll/_cci2/testing-ios.md

source 'https://github.com/CocoaPods/Specs.git'
```

#### Homebrew

Homebrew, by default, will check for updates at the start of any operation. As Homebrew has a fairly frequent release cycle, this means that the step execution can take some extra time to complete.

If build speed, or bugs introduced by new Homebrew updates, are of a concern, this can be disabled. On average, this can save 2-5 minutes per job.

To disable this feature, add the following line to your config.yml before calling Homebrew:

```yaml

> - run: echo "HOMEBREW_NO_AUTO_UPDATE=1" >> $BASH_ENV

Reviewer: My preference would be to set this as an env var in the job:

```
jobs:
  build:
    macos:
      xcode: 11.3.0
    environment:
      HOMEBREW_NO_AUTO_UPDATE: 1
    steps:
      - checkout
```

```

#### iOS Simulator Crash Reports

If your iOS app crashes in the Simulator during a test run, the crash report is useful for diagnosing the exact cause of the crash. These can be uploaded as artifacts to assist with debugging the app crash:

```yaml
- store_artifacts:
    path: ~/Library/Logs/DiagnosticReports

> ```

Reviewer: This is a nice tip 👍

source 'https://github.com/CocoaPods/Specs.git'
```

#### Homebrew

Homebrew, by default, will check for updates at the start of any operation. As Homebrew has a fairly frequent release cycle, this means that the step execution can take some extra time to complete.

> If build speed, or bugs introduced by new Homebrew updates, are of a concern, this can be disabled. On average, this can save 2-5 minutes per job.

Reviewer 2: ```suggestion
If build speed or bugs introduced by new Homebrew updates are a concern, this update-check feature can be disabled. On average, this can save 2-5 minutes per job.
```

> To disable this feature, define the `HOMEBREW_NO_AUTO_UPDATE` environment variable config.yml in your job:

Reviewer 2: ```suggestion
To disable this feature, define the `HOMEBREW_NO_AUTO_UPDATE` environment variable within your job:
```

```yaml
version: 2.1
jobs:
  build-and-test:
    macos:
      xcode: 11.3.0
    environment:
      HOMEBREW_NO_AUTO_UPDATE: 1
    steps:
      - checkout
      - run: brew install wget
```

#### iOS Simulator Crash Reports

> If your iOS app crashes in the Simulator during a test run, the crash report is useful for diagnosing the exact cause of the crash. These can be uploaded as artifacts to assist with debugging the app crash:

Reviewer 2: ```suggestion
If your iOS app crashes in the Simulator during a test run, the crash report is useful for diagnosing the exact cause of the crash. Crash reports can be uploaded as artifacts, as follows:
```

```yaml

> version: 2.1

Reviewer 2: I believe if we change this to 2.1, the `version 2` bit from the workflows section should be removed :-)
